src/store/authorSlice.ts

```diff
@@ -0,0 +1,23 @@
+import { User } from '../types/User';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+type AuthorState = {
+  author: User | null;
+};
+
+const initialState: AuthorState = {
+  author: null,
+};
+
+export const authorSlice = createSlice({
+  name: 'author',
+  initialState,
+  reducers: {
+    setAuthor(state, { payload }: PayloadAction<User>) {
+      // eslint-disable-next-line no-param-reassign
```

Comment:
[Not critical]
If you see that some eslint rule conflicts with parts of your code and/or if some eslint rule doesn't make development more comfortable and failsafe, I'd avaive to consider switching this rule off on config level rather than disabling it for certain lines pr files.